### PR TITLE
Forgive x,y outside grid in intersect method

### PR DESCRIPTION
**Pull request by briochh**
Added an option to forgive passing x,y coords that are outside the model grid to the `modelgrid.intersect()` method. 

This was motivated by many of my scripts failing after refactoring to use the intersect method instead of the previous dis.get_rc... method. I am aware that this is catering to my own laziness in not checking coords before trying to pull out the row and columns.

I've added to the tests too to catch and pass the exceptions thrown, as appropriate.

### CODE REVIEW

**Comment by rubencalje** [comment]
Goog option. It may be better to set forgive to True by default. Or was the change to an Exception (when the new grid-objects were made) intentional?

File: flopy/discretization/structuredgrid.py [hunk 1]
```diff
@@ -239,21 +241,32 @@ def intersect(self, x, y, local=False):
...
 
         # get the cell edges in local coordinates
         xe, ye = self.xyedges
 
         xcomp = x > xe
         if np.all(xcomp) or not np.any(xcomp):
-            raise Exception('x, y point given is outside of the model area')
-        col = np.where(xcomp)[0][-1]
+            if forgive:
+                col = np.nan
```
Comment (jdhughes-usgs) [inline flopy/discretization/structuredgrid.py]: I think None would be better than np.nan

File: flopy/discretization/structuredgrid.py [hunk 2]
```diff
@@ -239,21 +241,32 @@ def intersect(self, x, y, local=False):
...
+        else:
+            col = np.where(xcomp)[0][-1]
 
         ycomp = y < ye
         if np.all(ycomp) or not np.any(ycomp):
-            raise Exception('x, y point given is outside of the model area')
-        row = np.where(ycomp)[0][-1]
-
+            if forgive:
+                row = np.nan
```
Comment (jdhughes-usgs) [inline flopy/discretization/structuredgrid.py]: Suggest `None` instead of `np.nan`

File: flopy/discretization/structuredgrid.py [hunk 3]
```diff
@@ -239,21 +241,32 @@ def intersect(self, x, y, local=False):
...
+            else:
+                raise Exception(
+                    'x, y point given is outside of the model area')
+        else:
+            row = np.where(ycomp)[0][-1]
+        if np.any(np.isnan([row, col])):
```
Comment (jdhughes-usgs) [inline flopy/discretization/structuredgrid.py]: Suggest returning `None` for whichever failed. For example, `None, 16`, `10, None`, `None, None`.

File: flopy/discretization/vertexgrid.py [hunk 4]
```diff
@@ -165,6 +167,9 @@ def intersect(self, x, y, local=False):
                     radius = 1e-9
                 if path.contains_point((x, y), radius=radius):
                     return icell2d
+        if forgive:
+            icell2d = np.nan
```
Comment (jdhughes-usgs) [inline flopy/discretization/vertexgrid.py]: Suggest `None` instead of `np.nan`

**Comment by jwhite-usgs** [comment]
I think `np.nan` is actually a good choice, especially for those who use pandas since `np.nan` is tolerated on follow-on calculations: [https://stackoverflow.com/questions/17534106/what-is-the-difference-between-nan-and-none](https://stackoverflow.com/questions/17534106/what-is-the-difference-between-nan-and-none)

File: flopy/discretization/structuredgrid.py (hunk 3 above)
Comment (briochh) [reply flopy/discretization/structuredgrid.py]: Hi @jdhughes-usgs, can change to return `None` but `np.nan` is generally easier to work with downstream with both numpy and pandas (see @jwhite-usgs comment below) . Re. returning `NaN` (or `None`) for both, I guess was figuring that if is it outside grid for one it is outside the grid for both row and col and therefore shouldn't be assigned a value for either, happy to change to just return `NaN` (or `None`) for whichever is out of grid `x` or `y` range, if there is a strong preference for that.

**Pull request merged**